axelor-project/src/main/java/com/axelor/apps/project/service/SprintServiceImpl.java

```diff
@@ -39,4 +41,31 @@ public void generateBacklogSprint(ProjectVersion projectVersion) {
     Sprint sprint = new Sprint("Backlog - " + projectVersion.getTitle());
     projectVersion.setBacklogSprint(sprint);
   }
+
+  @Override
+  public boolean checkSprintOverlap(Project project) {
+    List<Sprint> sprintList = project.getSprintList();
+    if (CollectionUtils.isEmpty(sprintList)) {
+      return false;
+    }
+    return sprintList.stream()
+        .flatMap(
+            sprint ->
+                sprintList.stream()
+                    .filter(otherSprint -> !sprint.equals(otherSprint))
+                    .filter(otherSprint -> isOverlapping(sprint, otherSprint)))
+        .findFirst()
+        .isPresent();
+  }
+
+  protected boolean isOverlapping(Sprint sprint, Sprint otherSprint) {
+    if (sprint.getFromDate() == null
+        || sprint.getToDate() == null
+        || otherSprint.getFromDate() == null
+        || otherSprint.getToDate() == null) {
+      return false;
+    }
+    return sprint.getFromDate().isBefore(otherSprint.getToDate())
+        && sprint.getToDate().isAfter(otherSprint.getFromDate());
```

Comment:
It will only say that it's overlapping if sprint.fromDate < otherSprint.fromDate < otherSprint.toDate < sprint.toDate. 
We need to check that sprint.fromDate is between other sprint dates OR sprint.toDate is between other sprint dates indeed